symbolic-debuginfo/src/sourcebundle/utf8_reader.rs

```diff
@@ -0,0 +1,294 @@
+//! UTF-8 reader used by the sourcebundle module to read files.
+
+use std::io::{BufRead, Error, ErrorKind, Read, Result};
+use thiserror::Error;
+
+const MAX_UTF8_SEQUENCE_SIZE: usize = 4;
+const MAX_BUFFER_SIZE: usize = 8 * 1024;
+
+#[derive(Debug, Error)]
+#[error("Invalid UTF-8 sequence")]
+pub(crate) struct UTF8ReaderError;
+
+pub struct Utf8Reader<R> {
+    inner: R,
+
+    /// buffer_array always contains a valid UTF-8 sequence. It is possible that
+    /// buffer_vec[buffer_pos..] does not contain a valid UTF-8 sequence, if
```

Comment:
This still refers to `buffer_vec`.
```suggestion
    /// buffer_array[buffer_pos..] does not contain a valid UTF-8 sequence, if
```